vlog4j-graal/src/main/java/org/semanticweb/vlog4j/graal/GraalToVLog4JModelConverter.java

```diff
@@ -0,0 +1,214 @@
+	 *
+	 * @param predicate A {@link fr.lirmm.graphik.graal.api.core.Predicate Graal
+	 *                  Predicate}
+	 * @return A {@link Predicate VLog4J Predicate}
+	 */
+	private static Predicate convertPredicate(final fr.lirmm.graphik.graal.api.core.Predicate predicate) {
+		return makePredicate(predicate.getIdentifier().toString(), predicate.getArity());
+	}
+
+	/**
+	 * Converts a {@link ConjunctiveQuery Graal Query} into a
+	 * {@link ConjunctiveGraalQueryToRule}. Answering a {@link ConjunctiveQuery
+	 * GraalConjunctiveQuery} is equivalent to adding a {@link Rule} with the query
+	 * atoms as the body and a single atom with a new predicate containing all the
+	 * query variables as the head. This rule head can then be used as a query atom
+	 * to obtain the results of the query.
+	 *
+	 * <p>
+	 * <b>WARNING</b>: The supplied {@code ruleHeadPredicateName} will be used to
+	 * create a predicate containing all answer variables from the {@code query}. If
+	 * you use this name in another predicate, you will get conflicts.
+	 * </p>
+	 *
+	 * @param ruleHeadPredicateName A name to create a program-unique predicate for
+	 *                              the query atom.
+	 * @param query                 A {@link ConjunctiveQuery Graal Query}.
+	 * @return A {@link ConjunctiveGraalQueryToRule} equivalent to the {@code query}
+	 *         input.
+	 */
+	public static ConjunctiveGraalQueryToRule convertQuery(final String ruleHeadPredicateName, final ConjunctiveQuery query) {
+		final Conjunction conjunction = convertAtomSet(query.getAtomSet());
+		final List<Term> answerVariables = convertTerms(query.getAnswerVariables());
+
+		return new ConjunctiveGraalQueryToRule(ruleHeadPredicateName, answerVariables, conjunction);
+	}
+
+	/**
+	 * Converts a {@link fr.lirmm.graphik.graal.api.core.Rule Graal Rule} into a
+	 * {@link Rule Vlog4J Rule}.
+	 *
+	 * @param rule A {@link fr.lirmm.graphik.graal.api.core.Rule Graal Rule}.
+	 * @return A {@link Rule Vlog4J Rule}.
+	 */
+	public static Rule convertRule(final fr.lirmm.graphik.graal.api.core.Rule rule) {
+		final Conjunction head = convertAtomSet(rule.getHead());
+		final Conjunction body = convertAtomSet(rule.getBody());
+		return makeRule(head, body);
+	}
+
+	/**
+	 * Converts a {@link List} of {@link fr.lirmm.graphik.graal.api.core.Rule Graal
+	 * Rules} into a {@link List} of {@link Rule VLog4J Rules}.
+	 *
+	 * @param rules A {@link List} of {@link fr.lirmm.graphik.graal.api.core.Rule
+	 *              Graal Rules}.
+	 * @return A {@link List} of {@link Rule VLog4J Rules}.
+	 */
+	public static List<Rule> convertRules(final List<fr.lirmm.graphik.graal.api.core.Rule> rules) {
+		final List<Rule> result = new ArrayList<>();
+		for (final fr.lirmm.graphik.graal.api.core.Rule rule : rules) {
+			result.add(convertRule(rule));
+			;
+		}
+		return result;
+	}
+
+	/**
+	 * Converts a {@link fr.lirmm.graphik.graal.api.core.Term Graal Term} into a
+	 * {@link Term VLog4J Term}. Tests if the term is a
```

Comment:
Validates that, instead of "tests if"

Reply:
Done. Rewrote the comment so it does not describe the inner workings of the method.

Reply:
Whoops. Corrected